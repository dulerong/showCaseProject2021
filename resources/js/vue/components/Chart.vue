<template lang="pug">
apexchart(ref="chart" type="line" height="350" :options="chartOptions" :series="chartData")
</template>

<script>
export default {
  created() {
    console.log(this.$vuetify.theme.dark)
  },
  props: {
    chartData: {
      type: Array,
      required: true
    }
  },
  computed: {
    chartOptions() {
      const THEME_MODE = this.$vuetify.theme.dark ? 'dark' : 'light'
      return {
        theme: {
          mode: THEME_MODE
        },
        chart: {
          height: 350,
          type: 'line',
          dropShadow: {
            enabled: true,
            color: '#000',
            top: 18,
            left: 7,
            blur: 10,
            opacity: 0.2
          },
          toolbar: {
            show: false
          }
        },
        // colors: ['#77B6EA', '#545454'],
        // dataLabels: {
        //   enabled: true,
        // },
        stroke: {
          curve: 'smooth'
        },
        title: {
          text: this.$t("japanPopulation.chart.title"),
          align: 'left'
        },
        grid: {
          borderColor: '#e7e7e7',
          row: {
            colors: ['#f3f3f3', 'transparent'], // takes an array which will be repeated on columns
            opacity: 0.5
          },
        },
        markers: {
          size: 1
        },
        xaxis: {
          categories: [
            '1960', '1965', '1970', '1975', '1980', '1985', '1990', '1995', '2000', '2005', '2010', '2015', '2020', '2025', '2030', '2035', '2040', '2045'
          ],
          title: {
            text: this.$t("japanPopulation.chart.xAxis")
          }
        },
        yaxis: {
          title: {
            text: this.$t("japanPopulation.chart.yAxis")
          },
          labels: {
            formatter: val => val.toLocaleString()
          }
        },
        legend: {
          position: 'top',
          horizontalAlign: 'right',
          floating: true,
          offsetY: -25,
          offsetX: -5
        }
      }
    }
  },
}
</script>