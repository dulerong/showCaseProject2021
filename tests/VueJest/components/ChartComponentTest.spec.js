import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Chart from '~components/Chart'

describe('SettingDrawer Component', () => {
  const localVue = createLocalVue()

  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = (chartData = [], options) => {
    return mount(Chart, {
      localVue,
      vuetify,
      mocks: {
        $t: (msg) => msg,
      },
      propsData: {
        chartData
      },
      stubs: {
        ApexChart: true,
      },
      ...options
    })
  }

  it('Renders properly', () => {
    const wrapper = mountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Computed property: themeMode returns dark or light correctly', () => {
    const localThis = (darkMode) => {
      return {
        $vuetify: {
          theme: {
            dark: darkMode
          }
        }
      }
    }

    expect(Chart.computed.themeMode.call(localThis(false))).toEqual('light')

    expect(Chart.computed.themeMode.call(localThis(true))).toEqual('dark')
  })
})