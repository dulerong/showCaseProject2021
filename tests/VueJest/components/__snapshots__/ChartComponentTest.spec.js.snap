// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingDrawer Component Renders properly 1`] = `<apexchart-stub type="line" height="350" options="[object Object]" series=""></apexchart-stub>`;
